src/servers/src/http/authorize.rs

```diff
@@ -125,28 +127,80 @@ async fn authorize<B: Send + Sync + 'static>(
     user_provider.authorize(catalog, database, user_info).await
 }
 
+fn get_influxdb_credentials<B: Send + Sync + 'static>(
+    request: &Request<B>,
+) -> Result<Option<(Username, Password)>> {
+    // compat with influxdb v2 and v1
+    if let Some(header) = request.headers().get(http::header::AUTHORIZATION) {
+        // try v2 first
+        let (auth_scheme, credential) = header
+            .to_str()
+            .context(InvisibleASCIISnafu)?
+            .split_once(' ')
+            .context(InvalidAuthorizationHeaderSnafu)?;
+        ensure!(
+            auth_scheme.to_lowercase() == "token",
+            UnsupportedAuthSchemeSnafu { name: auth_scheme }
+        );
+
+        let (username, password) = credential
+            .split_once(':')
+            .context(InvalidAuthorizationHeaderSnafu)?;
+
+        Ok(Some((username.to_string(), password.to_string())))
+    } else {
+        // try v1
+        let query_str = request.uri().query();
+        if query_str.is_none() {
+            return Ok(None);
+        }
```

Comment:
Try this:

```suggestion
        let Some(query_str) = query_str else { return Ok(None) };
```

and remove the unwrap in line 159.

Reply:
we can simply `if let Some(query_str) = request.uri().query()` here